mongoengine/connection.py

```diff
@@ -177,10 +192,12 @@ def get_db(alias=DEFAULT_CONNECTION_NAME, reconnect=False):
         conn_settings = _connection_settings[alias]
         db = conn[conn_settings['name']]
         # Authenticate if necessary
-        if conn_settings['username'] and conn_settings['password']:
```

Comment:
Because when use MONGODB-X509 authentication mechanism not need specify password argument, only username.

http://api.mongodb.com/python/3.0b0/examples/authentication.html#mongodb-x509

While I agree that it is necessary to check the username for other authentication mechanisms.